imager/imager_images/views.py

```diff
@@ -1,3 +1,56 @@
-from django.shortcuts import render
+from django.core.urlresolvers import reverse_lazy
+from models import Photo, Albums
+from django.views.generic import CreateView, UpdateView
+# from django.contrib.auth.decorators import login_required
+from django.contrib.auth.views import redirect_to_login
+from django.shortcuts import redirect
+from profiles.models import ImagerProfile
 
-# Create your views here.
+class Add_Photo(CreateView):
+    model = Photo
+    fields = ['title', 'description', 'image', 'published']
+    template_name = 'photo_add.html'
+
+    def form_valid(self, form):
+        form.instance.profile = ImagerProfile.objects.get(
+            pk=self.request.user.profile.id)
+        return super(Add_Photo, self).form_valid(form)
+
+    success_url = reverse_lazy('library')
+
+
+class PhotoEditUpdateView(UpdateView):
+    model = Photo
+    fields = ['title', 'description', 'published']
+    template_name = 'photo_edit.html'
+    def dispatch(self, request, *args, **kwargs):
```

Comment:
If you'd done this in a more logical place, like the `get_object` method of the update view, you wouldn't have to add two extra queries to the view.  But by overriding `dispatch` you do so.